.github/workflows/clippy.yml

```diff
@@ -1,4 +1,6 @@
-on: push
+on:
+  push:
```

Comment:
we can probably exclude master here if we are running the job on pr

Reply:
```
on:
  push:
    branches-ignore:
      - 'master'
```

Reply:
While that's true we intentionally left it in for "our" own builds because if someone clicks on the green checkmark on the main repository he/she will see that clippy ran as well. More "marketing" than useful though.

Happy to remove it. I can push an update later.

Reply:
Oh, never considered that as a marketing point. I never click on that on master, I tend to to a PR and look at required statuses there (because that list is usually always bigger). I'm leaning more to limiting actions to points where it can give us actionable feedback, particularly since it's a finite free resource (2000min/mo)

Reply:
> particularly since it's a finite free resource (2000min/mo)

I believe it's free for public projects.

> GitHub Actions usage is free for both public repositories and self-hosted runners.
> https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions

Actions Usage is still 0 for this org.

Reply:
ohhh that's how it works. i thought that was a bug. :dumpling: